### ISSUE TITLE
Workflow build mess

### ISSUE THREAD

**Issue by KFlash** [cherow/cherow#255]
@fkleuver Before when we build cherow with both previous workflows the Rollup bundled files ended up in dist folder. With previous workflow the TS types ended up in a sub folder.

If you try to bundle now a lot of files ending up in the dist folder. No structure. 
And we also got two sub folders now - "lexer" and "parser" because it is build by TS itself. In previous workflows this kind of build was done to a build folder then bundled and put into the dist folder.

What happened?

**Comment by KFlash** [comment]
We also never had something like this before in the bundled output. It take too much space. How can we keep it small sized?

```ts
var Context;
(function (Context) {
    Context[Context["Empty"] = 0] = "Empty";
    Context[Context["OptionsRaw"] = 2] = "OptionsRaw";
    Context[Context["OptionsNext"] = 4] = "OptionsNext";
    Context[Context["OptionsLoc"] = 8] = "OptionsLoc";
    Context[Context["OptionsRanges"] = 16] = "OptionsRanges";
    Context[Context["OptionsJSX"] = 32] = "OptionsJSX";
    Context[Context["OptionsWebCompat"] = 64] = "OptionsWebCompat";
    Context[Context["OptionsGlobalReturn"] = 128] = "OptionsGlobalReturn";
    Context[Context["OptionsCollectComments"] = 256] = "OptionsCollectComments";
    Context[Context["OptionsExperimental"] = 512] = "OptionsExperimental";
    Context[Context["OptionsNative"] = 1024] = "OptionsNative";
    Context[Context["OptionsTokenize"] = 2048] = "OptionsTokenize";
    Context[Context["ExpressionStart"] = 4096] = "ExpressionStart";
    Context[Context["InGenerator"] = 8192] = "InGenerator";
    Context[Context["InAsync"] = 16384] = "InAsync";
    Context[Context["InParam"] = 32768] = "InParam";
    Context[Context["Strict"] = 65536] = "Strict";
    Context[Context["Module"] = 131072] = "Module";
    Context[Context["TaggedTemplate"] = 262144] = "TaggedTemplate";
    Context[Context["SkipDuplicates"] = 524288] = "SkipDuplicates";
    Context[Context["InClass"] = 1048576] = "InClass";
    Context[Context["NewTarget"] = 2097152] = "NewTarget";
    Context[Context["InFunctionBody"] = 4194304] = "InFunctionBody";
    Context[Context["DisallowIn"] = 8388608] = "DisallowIn";
    Context[Context["Global"] = 16777216] = "Global";
    Context[Context["DisallowGenerator"] = 33554432] = "DisallowGenerator";
    Context[Context["ScopeRoot"] = 67108864] = "ScopeRoot";
    Context[Context["Statement"] = 134217728] = "Statement";
})(Context || (Context = {}));

```

**Comment by futagoza** [comment]
Isn't that a normal output for TypeScript enums with string values?

**Comment by KFlash** [comment]
If we want to keep the values but we didn't do that before

**Comment by Aladdin-ADD** [comment]
@KFlash have you tried `npm run bundle`?

**Comment by KFlash** [comment]
Yes. Still messy

**Comment by fkleuver** [comment]
@KFlash You need to remove the const enums from the public api first. If you're exposing them in the public api then you need to preserve them, or people who try to use them will get `undefined` reference errors.

As for the structure, this one is simple:
Type definitions and mappings are kept as-is - their structure simply stays identical to the `src` folder structure. This is to prevent duplication and keep the mappings correct for a good debugging experience.
The actually imported files (those that have `.system`, `.amd` suffixes etc) are plainly placed in the root level along with their minified equivalents.

I understand it seems messy if you don't understand the outputs. Hopefully you understand now and see that this is in fact very clean for how complete it is.

**Comment by KFlash** [comment]
@fkleuver Are we actually going to expose the const enums now? We did it before and ran into issues but this time around I thought we should avoid it? That's one of the reasons I'm now using more than two arguments too. Before we only passed `state` and `context`, but now also we got `scope` and probably soon we get a few more. And even more const enums to keep track of. E.g. passing `BindingOrigin` and `BindingType`. Both have their own const enums and func arg. 

You haven't seen this code yet, but I just mentioned it...

Rest of the workflow make sense as long as it is small size. Main goal is to be 20KB less than Acorn as v. 1.6.8 is.
